src/siwe/siweVerification.ts

```diff
@@ -0,0 +1,66 @@
+import { SiweMessage, SiweError } from 'siwe';
+import { z } from 'zod';
+
+// import { DOMAIN } from '../constants';
+
+import type { json } from './jsonSchema';
+
+const stringToJSONSchema = z.string().transform((str, ctx): z.infer<ReturnType<typeof json>> => {
+  try {
+    return JSON.parse(str);
+  } catch (e) {
+    ctx.addIssue({ code: 'custom', message: 'Invalid JSON' });
+    return z.NEVER;
+  }
+});
+
+const signedMessageSchema = z.object({
+  message: z.string(), // NOT JSON -- a SIWE schema message from `prepareMessage(), which must be parsed by `SiweMessage`
+  signature: z.string(),
+});
+
+async function validateSIWEMessage(rawSignedMessage: string): Promise<string> {
+  const parsedSignedMessage = stringToJSONSchema.parse(rawSignedMessage);
+  const { message, signature } = signedMessageSchema.parse(parsedSignedMessage);
+
+  const siweMessage = new SiweMessage(message);
+
+  // Verify using the static verify method of SiweMessage
+  const { error, success } = await siweMessage.verify({
+    signature,
+    // TODO: Add domain verification
+    // domain: DOMAIN,
+  });
```

Comment:
Where are these SIWE msgs used by all the endpoints coming from?

Are users building it themselves? In that case, we need a `getSIWE` endpoint where we define the SIWE msg we want them to sign and temporarily store it for proper validation when they use it.

Without it they can attack our validation system by altering the nonce or timestamps (which btw I do not see us properly enforcing `expiredAt`)
As it is currently, any siwe msg they create and sign will work. It might be un-expirable and even craftable by looping the nonce to make it match someone else's signature